utils/puppeteerScrapper.js

```diff
@@ -1,151 +1,90 @@
 const puppeteer = require('puppeteer'); // v13.0.0 or later
 
-const getWaifu = async () => {
-  const waifu = {
-    name: '',
-    imageUrl: '',
-    bioUrl: '',
-    extract: '',
-    appearsIn: '',
+const nameSelector = 'h1';
+const imageSelector = '#widget-waifu-of-the-day > a > img';
+const popularitySelector = '#popularity-rank';
+const ageSelector = '#age';
+const husbandoSelector = '#waifu-classification';
+const appearsSelector = '#waifu-core-information > div > div > a';
+
+const waifuURL = "https://mywaifulist.moe/"
+const randomUrl = "https://mywaifulist.moe/random"
+
+const getWaifu = async (retry = false) => {
+  const [browser, page] = await intializeScrapper()
+  await page.setViewport({ "width": 1920, "height": 1080 })
+
+  const url = retry ? randomUrl : waifuURL
+  console.info(`Navigating to: ${url}`)
+  await page.goto(url, {waitUntil: "networkidle2"});
+
+  if (!retry) {
+    console.info("Looking for WOTD section...");
+    const section = await page.waitForSelector(imageSelector)
+
+    console.info("Loading waifu info");
+    await section.click()
+    await page.waitForNavigation({waitUntil: "networkidle2"})
   }
 
+  console.info("Detecting Husbandos...");
+  const isHusbando = await page.$(husbandoSelector)
 
-  const hrefSelector = '#widget-waifu-of-the-day > a';
-  const nameSelector = '#widget-waifu-of-the-day .text-zinc-100';
-  const imageSelector = '#widget-waifu-of-the-day > a > img';
-  const descriptionSelector = '#description';
-  const appearsSelector = '#waifu-core-information > div > div > a';
+  if (isHusbando) {
+    console.info("Husbando detected, retry a random waifu...")
+    await browser.close();
+    return await getWaifu(true)
+  }
```

Comment:
_This is just a suggestion_

This logic can be outside the `getWaifu` function and inside the `app.js` file instead. Why you might want that? Well, having in the stack two different instances of browser/puppeteer instances running at the same time may be overkill and also may cause issues if for some reason you get husbandos several times in row. A better approach would be to just give the `getWaifu` function the task to get a character then you validate that outside the function so the browser/pupeteer instance is disposed before trying again with another URL, after that you can check if it's a husbando or not and then decide if you want to try now with the random URL instead.

You can also terminate the `getWaifu` function early by returning `null` (and disposing the browser/puppeteer instance accordingly) indicating that no waifu was found, so you can have the "keep looking" logic in the `app.js` file.

Reply:
Counsel has spoken, husbando/retry functionality will be removed